projects/filigran-website/components/display-all-icons.tsx

```diff
@@ -1,16 +1,48 @@
+'use client'
 import * as FiligranIcon from 'filigran-icon'
-import {Fragment} from 'react'
+import React, {Fragment,ReactElement,useState,useMemo} from 'react'
+import Popup from './icon-popup';
+import SearchBar from './icon-searchbar';
 
 export const DisplayAllIcons = () => {
+  const [selectedIcon, setSelectedIcon] = useState<ReactElement | null>(null);
+  const [selectedIconLabel, setSelectedIconLabel] = useState('');
+  const [isOpen, setIsOpen] = useState(false);
+  const [searchTerm, setSearchTerm] = useState('');
+
+  const handleIconClick = (Icon : React.ComponentType<any> ,icon : string) => {
+    setSelectedIcon(<Icon className="w-20 h-20"/>);
+    setSelectedIconLabel(icon);
+    setIsOpen(true);
+  }
+
+  const handleClose = () => {
+  setIsOpen(false);
+  setSelectedIcon(null);
+  setSelectedIconLabel('');
+  } 
+
   const allIcons = Object.keys(FiligranIcon).filter((key) => key !== 'default')
```

Comment:
My mistake, but we can place the allIcons outside the component since we don't need to rerender it every time. We should also remove it from the dependencies of useMemo